File: src/main/java/domain/stock/Holding.java
package domain.stock;

//===========================================================//
/**
 * Represents a stock holding in a portfolio.
 *
 * @param entryPrice - the average purchase price per share.
 * @param amount - the number of shares held.
 */
//===========================================================//

public record Holding(
    double entryPrice,
    long amount
) {
    public Holding {
        if(entryPrice <= 0d) throw new IllegalArgumentException("Price");
        if(amount == 0L) throw new IllegalArgumentException("Amount");
    }
}

File: src/main/kotlin/domain/stock/Holding.kt
package domain.stock

//===========================================================//

/**
 * Represents a stock holding in a portfolio.
 * 
 * @param entryPrice - the average purchase price per share.
 * @param amount - the number of shares held.
 */
//===========================================================//

data class Holding(
    val entryPrice: Double,
    val amount: Long
) {
    init {
        require(entryPrice >= 0.0) { "Price" }
        require(amount != 0L) { "Amount" }
    }
}
